### PR TITLE
ci: update to Ubuntu-22.04-XL runners

### PR DESCRIPTION
**Pull request by littledivy**
Updates CI to use ubuntu-22.04-xl runners. A change to sysroot setup
was necessary that links `libdl.so.2` and `libdl.a` that are no longer
present on ubuntu-22.04.

### CODE REVIEW

File: .github/workflows/ci.generate.ts [hunk 1]
```diff
@@ -81,7 +86,7 @@ const submoduleStep = (submodule: string) => ({
 });
 
 const installRustStep = {
-  uses: "dsherret/rust-toolchain-file@v1",
```
Comment (bartlomieju) [inline .github/workflows/ci.generate.ts:84]: This should be reverted

File: .github/workflows/ci.generate.ts [hunk 2]
```diff
@@ -473,12 +478,6 @@ const ci = {
             run:
               "deno run --unstable --allow-write --allow-read --allow-run ./tools/format.js --check",
           },
-          {
-            name: "Lint PR title",
-            if: "matrix.job == 'lint' && github.event_name == 'pull_request'",
-            run:
-              "deno run ./tools/verify_pr_title.js '${{ github.event.pull_request.title }}'",
-          },
```
Comment (bartlomieju) [inline .github/workflows/ci.generate.ts:481]: Ditto

**Comment by bartlomieju** [comment]
@littledivy are `libdl.so` and `libdl.a` files are needed here?

**Comment by littledivy** [comment]
@bartlomieju yeah those are needed, they are from ubuntu 20's `libc6-dev`. I'll add them to deno_third_party

**Comment by littledivy** [comment]
It works.
```
[ec2-user@ip-172-31-53-34 ~]$ objdump -T ./deno | grep -E 'GLIBC_2.[23][0-9]'
[ec2-user@ip-172-31-53-34 ~]$ ./deno
Deno 1.31.2
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> 
[ec2-user@ip-172-31-53-34 ~]$ 
```

File: .github/workflows/ci.generate.ts [hunk 3]
```diff
@@ -43,6 +43,9 @@ sudo mount --rbind /sys /sysroot/sys
 sudo mount --rbind /home /sysroot/home
 sudo mount -t proc /proc /sysroot/proc
 
+sudo ln -s third_party/prebuilt/linux64/libdl/libdl.so.2 /sysroot/lib/x86_64-linux-gnu/libdl.so
```
Comment (bartlomieju) [inline .github/workflows/ci.generate.ts:46]: Shouldn't it be `third_party/prebuilt/linux64/libdl/libdl.so`?

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**